src/parquet/column/reader.cc

```diff
@@ -190,17 +196,17 @@ static size_t DecodeMany(RleDecoder* decoder, int16_t* levels, size_t batch_size
 }
 
 size_t ColumnReader::ReadDefinitionLevels(size_t batch_size, int16_t* levels) {
-  if (!definition_level_decoder_) {
+  if (!definition_level_decoder_.set()) {
```

Comment:
I think it would be better to check whether the `max_definition_level_` is greater than 0 here. The contract about the existence of the definition level decoder should be enforced elsewhere